### PR TITLE
docs(x/poolmanager): module README

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #3881

## What is the purpose of the change

This PR adds a README for the `poolmanager` module.

This README is extracted from the [concentrated-liquidity architecture](https://github.com/osmosis-labs/osmosis/blob/4a81ee507561a5296fdeaf5f6905dd030d36d9c8/x/concentrated-liquidity/architecture.md#L459) as well as the GAMM module spec.

The CL architecture file was also updated.

Obsolete documentation from `x/gamm/README.md` was removed. Linked to `poolmanager` instead.

### CODE REVIEW

File: x/concentrated-liquidity/architecture.md [hunk 1]
```diff
@@ -466,10 +453,12 @@ enum PoolType {
 ```
 
 Let's begin by considering the execution flow of the pool creation message.
+Assume `balancer` pool is being created.
 
-1. `CreatePoolMsg` is received by the `poolmanager` message server.
+1. `CreatePoolMsg` is received by the `x/gamm` message server.
 
-2. `CreatePool` `poolmanager` keeper method is called.
+2. `CreatePool` `poolmanager` keeper method is called, propagating
```
Comment (mattverse) [inline x/concentrated-liquidity/architecture.md:460]:
```suggestion
2. `CreatePool` keeper method is called from `poolmanager` module, propagating
```

File: x/poolmanager/README.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+To make sure that the pool ids are unique across the two modules, we unify pool id management
+in the `poolmanager`.
+
+When a call to `CreatePool` keeper method is received, we get the next pool id from the module
+storage, assign it to the new pool and propagate the execution to either `gamm`
+or `concentrated-liquidity` modules.
+
+Note that we define a `CreatePoolMsg` interface:
+<https://github.com/osmosis-labs/osmosis/blob/main/x/poolmanager/types/msg_create_pool.go#L9>
```
Comment (mattverse) [inline x/poolmanager/README.md:44]: ditto here for code lines

File: x/poolmanager/README.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+  Concentrated = 2;
+}
+```
+
+Let's begin by considering the execution flow of the pool creation message.
+Assume `balancer` pool is being created.
+
+1. `CreatePoolMsg` is received by the `x/gamm` message server.
+
+2. `CreatePool` `poolmanager` keeper method is called, propagating
```
Comment (mattverse) [inline x/poolmanager/README.md:76]:
```suggestion
2. `CreatePool` keeper method is called  from `poolmanager`, propagating
```

File: x/poolmanager/README.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+	tokenIn sdk.Coin,
+	tokenOutMinAmount sdk.Int) (tokenOutAmount sdk.Int, err error) {
+}
+```
+
+Essentially, the method iterates over the routes and calls a `SwapExactAmountIn` method
+for each, subsequently updating the inter-pool swap state.
+
+The routing works by querying the index `SwapModuleRouterPrefix`,
+searching up the `poolmanagerkeeper.router` mapping, and callig
```
Comment (mattverse) [inline x/poolmanager/README.md:213]:
```suggestion
searching up the `poolmanagerkeeper.router` mapping, and calling
```

File: x/poolmanager/README.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+Given a `tokenIn`, the following calculations are done to calculate how
+many tokens are to be swapped into and removed from the pool:
+
+`tokenBalanceOut * [1 - { tokenBalanceIn / (tokenBalanceIn + (1 - swapFee) * tokenAmountIn)} ^ (tokenWeightIn / tokenWeightOut)]`
+
+The calculation is also able to be reversed, the case where user
+provides `tokenOut`. The calculation for the amount of tokens that the
+user should be putting in is done through the following formula:
+
+`tokenBalanceIn * [{tokenBalanceOut / (tokenBalanceOut - tokenAmountOut)} ^ (tokenWeightOut / tokenWeightIn) -1] / tokenAmountIn`
```
Comment (mattverse) [inline x/poolmanager/README.md:263]: Isnt this only for Balancer?

File: x/poolmanager/README.md [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+
+## Messages
+
+### MsgSwapExactAmountIn
+
+[MsgSwapExactAmountIn](https://github.com/osmosis-labs/osmosis/blob/main/proto/osmosis/gamm/v1beta1/tx.proto#L85)
+
+### MsgSwapExactAmountOut
+
+[MsgSwapExactAmountOut](https://github.com/osmosis-labs/osmosis/blob/main/proto/osmosis/gamm/v1beta1/tx.proto#L108)
```
Comment (mattverse) [inline x/poolmanager/README.md:277]: ditto for code line

**Review by mattverse — APPROVED** [review]
Left some minor comments and feedbacks on some fronts, otherwise LGTM! Nice work on adding detailed docs, this would be super useful

File: x/poolmanager/README.md (hunk 5 above)
Comment (p0mvn) [reply x/poolmanager/README.md:264]: updated to be more specific

File: x/concentrated-liquidity/architecture.md [hunk 7]
```diff
@@ -416,29 +420,12 @@ Let's consider pool creation and swaps separately and in more detail.
...
-	PoolCreator() sdk.AccAddress
-	// A stateful validation function.
-	Validate(ctx sdk.Context) error
-	// Initial Liquidity for the pool that the sender is required to send to the pool account
-	InitialLiquidity() sdk.Coins
-	// CreatePool creates a pool implementing PoolI, using data from the message.
-	CreatePool(ctx sdk.Context, poolID uint64) (gammtypes.TraditionalAmmInterface, error)
-}
-```
+<https://github.com/osmosis-labs/osmosis/blob/main/x/poolmanager/types/msg_create_pool.go#L9>
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply x/concentrated-liquidity/architecture.md:428]: updated all links

**Pull request merged**